File: content/atlas/source/includes/fts/synonyms/equivalent-advanced-query.py
import pymongo
import dns

client = pymongo.MongoClient('<connection-string>')
result = client['sample_mflix']['movies'].aggregate([
    {
        '$search': {
            'index': 'synonyms-tutorial',
            'compound': {
                'should': [
                    {
                        'text': {
                            'path': 'title',
                            'query': 'automobile',
                            'synonyms': 'transportSynonyms'
                        }
                    }, {
                        'text': {
                            'path': 'title',
                            'query': 'attire',
                            'synonyms': 'attireSynonyms'
                        }
                    }
                ]
            }
        }
    }, {
        '$limit': 10
    }, {
        '$project': {
            '_id': 0,
            'title': 1,
            'score': {
                '$meta': 'searchScore'
            }
        }
    }
])

for i in result:
    print(i)

File: content/atlas/source/includes/fts/synonyms/equivalent-advanced-query.kt
import com.mongodb.client.model.Aggregates.limit
import com.mongodb.client.model.Aggregates.project
import com.mongodb.client.model.Projections.*
import com.mongodb.kotlin.client.coroutine.MongoClient
import kotlinx.coroutines.runBlocking
import org.bson.Document

fun main() {
    // establish connection and set namespace
    val uri = "<connection-string>"
    val mongoClient = MongoClient.create(uri)
    val database = mongoClient.getDatabase("sample_mflix")
    val collection = database.getCollection<Document>("movies")

    runBlocking {
        // define query
        val agg = Document(
            "\$search",
            Document("index", "synonyms-tutorial")
                .append(
                    "compound",
                    Document(
                        "should", listOf(
                            Document(
                                "text",
                                Document("path", "title")
                                    .append("query", "automobile")
                                    .append("synonyms", "transportSynonyms")
                            ),
                            Document(
                                "text",
                                Document("path", "title")
                                    .append("query", "attire")
                                    .append("synonyms", "attireSynonyms")
                            )
                        )
                    )
                )
        )

        // run query and print results
        val resultsFlow = collection.aggregate<Document>(
            listOf(
                agg,
                limit(10),
                project(fields(
                    excludeId(),
                    include("title"),
                    computed("score", Document("\$meta", "searchScore"))
                ))
            )
        )
        resultsFlow.collect { println(it) }
    }
    mongoClient.close()
}

